src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -396,10 +395,13 @@ private static ProjectDependency createConfigurationDependency(Project project,
     }
 
     /** Create a dependency to {@code toConfiguration}, where the latter should exist in the given {@code project}. */
-    private static Dependency createConfigurationDependencyWithScope(
-            Project project, Configuration toConfiguration, GcvScope scope) {
-        ModuleDependency dep = GradleWorkarounds.fixAttributesOfModuleDependency(
-                project.getObjects(), createConfigurationDependency(project, toConfiguration));
+    private static Dependency createConfigurationDependencyWithScope(Project project, GcvScope scope) {
+        ProjectDependency projectDependency = (ProjectDependency) project.getDependencies().create(project);
+        projectDependency.capabilities(moduleDependencyCapabilitiesHandler -> {
```

Comment:
This method and the handler have been introduced in 5.3 only:
https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/ModuleDependency.html#capabilities-org.gradle.api.Action-

Reply:
Added that to the changelog